modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobStore.java

```diff
@@ -360,30 +365,80 @@ public void deleteBlobsIgnoringIfNotExists(OperationPurpose purpose, Iterator<St
+        int retryCounter = 0;
+        while (true) {
+            try (AmazonS3Reference clientReference = clientReference()) {
+                SocketAccess.doPrivilegedVoid(() -> clientReference.client().deleteObjects(bulkDelete(purpose, this, partition)));
+                s3RepositoriesMetrics.retryDeletesHistogram().record(retryCounter);
+                return;
+            } catch (MultiObjectDeleteException e) {
+                // We are sending quiet mode requests so we can't use the deleted keys entry on the exception and instead
+                // first remove all keys that were sent in the request and then add back those that ran into an exception.
+                logger.warn(
+                    () -> format(
+                        "Failed to delete some blobs %s",
+                        e.getErrors()
+                            .stream()
+                            .map(err -> "[" + err.getKey() + "][" + err.getCode() + "][" + err.getMessage() + "]")
+                            .toList()
+                    ),
+                    e
+                );
+                aex.set(ExceptionsHelper.useOrSuppress(aex.get(), e));
+                return;
+            } catch (AmazonClientException e) {
+                if (shouldRetryDelete(purpose) && RetryUtils.isThrottlingException(e)) {
+                    // S3 is asking us to slow down. Pause for a bit and retry
+                    if (maybeDelayAndRetryDelete(retries)) {
+                        retryCounter++;
+                    } else {
+                        s3RepositoriesMetrics.retryDeletesHistogram().record(retryCounter);
+                        aex.set(ExceptionsHelper.useOrSuppress(aex.get(), e));
+                        return;
+                    }
+                } else {
+                    // The AWS client threw any unexpected exception and did not execute the request at all so we do not
+                    // remove any keys from the outstanding deletes set.
+                    aex.set(ExceptionsHelper.useOrSuppress(aex.get(), e));
+                    return;
+                }
+            }
+        }
+    }
+
+    /**
+     * If there are remaining retries, pause for the configured interval then return true
+     *
+     * @param retries The retries iterator
+     * @return true to try the deletion again, false otherwise
+     */
+    private boolean maybeDelayAndRetryDelete(Iterator<TimeValue> retries) {
+        if (retries.hasNext()) {
+            try {
+                Thread.sleep(retries.next().millis());
+                return true;
+            } catch (InterruptedException iex) {
+                Thread.currentThread().interrupt();
+                // If we're interrupted, record the exception and abort retries
+                logger.warn("Aborting delete retries due to interrupt");
+            }
+        } else {
+            logger.warn("Exceeded maximum delete retries, aborting");
```

Comment:
Could we have some more detail in this message about what exactly we're aborting, and the retry strategy we were using?

Reply:
Fixed in 52a315bca01e70101697969ae124769009795f3d

Reply:
Sorry I meant more details like the number of times we retried, and refs to the settings that can be used to influence this and/or a link to the relevant docs. Otherwise IME we eventually get support cases asking for that information.

Which reminds me, we don't document these new settings either, but we should.

Reply:
I had a go at this in 3ed3bc4da39377f16715762bfc2980de078c313d